azure/src/main/java/org/apache/iceberg/azure/AzureProperties.java

```diff
@@ -93,7 +93,7 @@ public void applyClientConfiguration(String account, DataLakeFileSystemClientBui
     if (connectionString != null && !connectionString.isEmpty()) {
       builder.endpoint(connectionString);
     } else {
-      builder.endpoint("https://" + account);
+      builder.endpoint("https://" + account + ".dfs.core.windows.net");
```

Comment:
Should we really change add `.def.core.windows.net` by default ?
I mean that use currently include the suffix in `account` so it should be clearly state that we append the suffix by default now.

Reply:
I think this is the right thing to do, I would just throw a javadoc on this method though

Reply:
This deviates from the `abfs[s]` scheme defined by Hadoop, where the domain is specified in the URI ([docs](https://hadoop.apache.org/docs/stable/hadoop-azure/abfs.html)). I feel we should check if the account already ends with the domain and only append then to support that.